mypy/newsemanal/semanal_main.py

```diff
@@ -17,6 +17,9 @@
 
 def semantic_analysis_for_scc(graph: 'Graph', scc: List[str]) -> None:
     # Assume reachability analysis has already been performed.
+    #
+    # Note that functions can't define new module-level attributes,
+    # since module top levels are fully processed before functions.
```

Comment:
Sorry for being annoying, but maybe explicitly mention `global x` here? (and clarify whether it is a temporary or principal limitation?)